viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -579,77 +579,55 @@ void SLAMServiceImpl::GetLatestSampledPointCloudMapString(
         }
     }
 
+    // Get data from painted surface in RGBA format
     auto painted_surface = painted_slices->surface.get();
     int width = cairo_image_surface_get_width(painted_surface);
     int height = cairo_image_surface_get_height(painted_surface);
-    // Get all pixels from the painted surface in RGBA format
     auto data = cairo_image_surface_get_data(painted_surface);
```

Comment:
[nit] Please use the type instead of `auto` for improved readability. I recommend [opt] to rename `data` to `surface_data` as well.

Reply:
I agree about wanting to be type-rigid in our code but not only does cartographer uses `auto` heavily, particularly when it comes to their custom types and `cairo` types but that is our current pattern as well. I personally think it's better to stick to both current patterns and bring up the topic of refactoring for type rigidity during full modularization.

I do agree though that the `rotated_map_point` should be defined since `Eigen:Vector3d` is a much more ubiquitous type.